File: processor/src/test/resources/scenarios/cassandracodecs/NonPublicCassandraCodec.java
package scenarios.cassandracodecs;

import com.datastax.oss.driver.api.core.type.codec.MappingCodec;
import com.datastax.oss.driver.api.core.type.codec.TypeCodecs;
import com.datastax.oss.driver.api.core.type.reflect.GenericType;

class NonPublicCassandraCodec extends MappingCodec<String, String> {

    public NonPublicCassandraCodec() {
        super(TypeCodecs.TEXT, GenericType.of(String.class));
    }

    @Override
    protected String innerToOuter(String value) {
        return value;
    }

    @Override
    protected String outerToInner(String value) {
        return value;
    }
}

File: ksp/src/test/resources/scenarios/cassandracodecs/NonPublicCassandraCodec.kt
package scenarios.cassandracodecs

import com.datastax.oss.driver.api.core.type.codec.MappingCodec
import com.datastax.oss.driver.api.core.type.codec.TypeCodecs
import com.datastax.oss.driver.api.core.type.reflect.GenericType

internal class NonPublicCassandraCodec :
    MappingCodec<String, String>(TypeCodecs.TEXT, GenericType.of(String::class.java)) {
    override fun innerToOuter(value: String?): String? = value
    override fun outerToInner(value: String?): String? = value
}
